epoch.go

```diff
@@ -897,12 +1045,39 @@ func (e *Epoch) locateBlock(seq uint64, digest []byte) (Block, bool) {
 	return nil, false
 }
 
-func (e *Epoch) proposeBlock() error {
-	block, ok := e.BlockBuilder.BuildBlock(e.finishCtx, e.Metadata())
-	if !ok {
-		return errors.New("failed to build block")
+func (e *Epoch) buildBlock() {
+	metadata := e.metadata()
+
+	task := e.createBlockBuildingTask(metadata)
+
+	// We set the task ID to be the hash of the previous block, since we don't know the digest
+	// of the block before it is built.
+	// We know, however, that any block before or after the previous block won't
+	// have this digest, under the assumption that the hash function is collision resistant.
+	// TODO: Inject this hash as a dependency and don't use SHA256 as a hardcoded function
+	taskID := sha256.Sum256(metadata.Prev[:])
```

Comment:
I find the introduction of a "fake" `taskID` pretty unusual tbh. I think we could defer the expectation of a taskID until _after_ the task has been executed. Then we could correctly use the blockID after it was built as the taskID.

Reply:
Changed it [here](https://github.com/ava-labs/Simplex/pull/38/commits/6763e9787581dd72c49fa62a2e00c0f75fe4ba14).